Automattic/wp-calypso
# Signup: add FSE to /start flow

#### Changes proposed in this Pull Request

Adds initial support for FSE to the /start flow.

- Uses an assignment function for easily changing the percentage of users that have new sites created as FSE sites
- Adds an additional callback to the design picker step that determines if FSE is active, once the theme is set up
- Redirects to the Site editor when the signup flow is complete (unless the "Write" intent is selected)
- Logs a tracks event `calypso_fse_enrolled` when a user creates a site and property site_enrolled: true/false, depending on if the users/site is enrolled in FSE.

#### Testing instructions

* Visit /start and create a site
* At the intent step, select "Build"
* In the Design picker, select a design that is based on Blockbase, such as Quadrat, Geologist, Zoologist, etc
* You should be redirected to the Site Editor after site setup is complete
* You should see FSE beta themes in Appearance themes

Note that if FSE isn't active for any reason (like the current theme doesn't support it), you won't be directed to the Site editor after setup, **but** you still should see the FSE treatment of the themes showcase because the site is eligible for FSE.

Related to #60255
